macOS and os_get_time_monotonic changes

kiwi64ajs opened the pull request:
added macOS specific support to the libify.sh shell script
changed os_get_time_monotonic() to just call the Arduino micros() function to improve precision and added a prototype to suppress the warning

balazsracz commented on src/os/os.c:
```diff
@@ -618,17 +618,10 @@ long long os_get_time_monotonic(void)
+	// redeclare micros() prototype to keep the compiler happy
+	unsigned long micros();
+	
+	time = micros();
```
1) you definitely need to keep the overflow handling for microseconds because those will overflow rather sooner than millis.
2) time is in nanoseconds so you need to do time *= 1000.

balazsracz commented on arduino/libify.sh:
```diff
@@ -92,7 +119,13 @@ function copy_dir() {
+
+    if [[ "$OSTYPE" == "darwin"* ]]; then
+	    cp -fa ${USE_LINK} ${OPENMRNPATH}/$2 .
+	else
```
please use spaces instead of tabs, because the indentation will be messed up.

kiwi64ajs commented:
Ok, I've:
- changed the tags to 4 spaces
- reinstated the micros() overflow detection
- converted micros() to uint_32
- updated last_micros after comparison to new_micros as it wasn't actually being saved
